pkg/asset/installconfig/nutanix/validation.go

```diff
@@ -35,24 +39,61 @@ func ValidateForProvisioning(ic *types.InstallConfig) error {
 		p.PrismCentral.Username,
 		p.PrismCentral.Password)
 	if err != nil {
-		return field.InternalError(field.NewPath("platform", "nutanix"), errors.Wrapf(err, "unable to connect to Prism Central %q", p.PrismCentral.Endpoint.Address))
+		return field.Invalid(parentPath.Child("prismCentral"), p.PrismCentral.Endpoint.Address, fmt.Sprintf("failed to connect to the prism-central with the configured credentials. %v", err))
 	}
 
 	// validate whether a prism element with the UUID actually exists
 	for _, pe := range p.PrismElements {
 		_, err = nc.V3.GetCluster(pe.UUID)
 		if err != nil {
-			return field.InternalError(field.NewPath("platform", "nutanix", "prismElements"), errors.Wrapf(err, "prism element UUID %s does not correspond to a valid prism element in Prism", pe.UUID))
+			errList = append(errList, field.Invalid(parentPath.Child("prismElements"), pe.UUID, fmt.Sprintf("the prism element %s's UUID does not correspond to a valid prism element in Prism. %v", pe.Name, err)))
 		}
 	}
 
 	// validate whether a subnet with the UUID actually exists
 	for _, subnetUUID := range p.SubnetUUIDs {
 		_, err = nc.V3.GetSubnet(subnetUUID)
 		if err != nil {
-			return field.InternalError(field.NewPath("platform", "nutanix", "subnetUUIDs"), errors.Wrapf(err, "subnet UUID %s does not correspond to a valid subnet in Prism", subnetUUID))
+			errList = append(errList, field.Invalid(parentPath.Child("subnetUUIDs"), subnetUUID, fmt.Sprintf("the subnet UUID does not correspond to a valid subnet in Prism. %v", err)))
 		}
 	}
 
-	return nil
+	if len(p.FailureDomains) > 0 {
+		// validate each FailureDomain configuration
+		for _, fd := range p.FailureDomains {
```

Comment:
No need for the `if`, if `len(p.FailureDomains) == 0` then the for won't do anything.
```suggestion
		// validate each FailureDomain configuration
		for _, fd := range p.FailureDomains {
```

Reply:
Fixed.